drivers/mtd/mtd_nvs.c

```diff
@@ -0,0 +1,2241 @@
+static ssize_t flash_write(struct mtd_dev_s *mtd, off_t offset,
+                           const uint8_t *data, size_t len);
+static int flash_erase(struct mtd_dev_s *mtd, off_t offset, size_t len);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static const struct file_operations mtdnvs_fops =
+{
+  mtdnvs_open,  /* open */
+  mtdnvs_close, /* close */
+  mtdnvs_read,  /* read */
+  NULL,         /* write */
+  NULL,         /* seek */
+  mtdnvs_ioctl, /* ioctl */
+  mtdnvs_poll   /* poll */
+#ifndef CONFIG_DISABLE_PSEUDOFS_OPERATIONS
+  , NULL        /* unlink */
+#endif
+};
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+uint32_t fnv_32_str(const char *str)
+{
+  unsigned char *s = (unsigned char *)str;
+  uint32_t hval = 2166136261;
+
+  /* FNV-1 hash each octet in the buffer */
+
+  while (*s)
+    {
+      /* multiply by the 32 bit FNV magic prime mod 2^32 */
+
+      hval *= 0x01000193;
+
+      /* xor the bottom with the current octet */
+
+      hval ^= (uint32_t)*s++;
+    }
+
+  return hval;
+}
+
+#ifdef CONFIG_MTD_NVS_LOOKUP_CACHE
+
+static inline size_t nvs_lookup_cache_pos(uint32_t id)
+{
+  size_t pos;
+
+#if CONFIG_NVS_LOOKUP_CACHE_SIZE <= UINT8_MAX
+  /* CRC8-CCITT is used for ATE checksums and it also acts well as a hash
+   * function, so it can be a good choice from the code size perspective.
+   * However, other hash functions can be used as well if proved better
+   * performance.
+   */
+
+  pos = crc8_ccitt(CRC8_CCITT_INITIAL_VALUE, &id, sizeof(id));
+#else
+  pos = crc16_ccitt(0xffff, (const uint8_t *)&id, sizeof(id));
+#endif
+
+  return pos % CONFIG_NVS_LOOKUP_CACHE_SIZE;
+}
+
+static int nvs_lookup_cache_rebuild(struct nvs_fs *fs)
+{
+  int rc;
+  uint32_t addr;
+  uint32_t ate_addr;
+  uint32_t *cache_entry;
+  struct nvs_ate ate;
+
+  memset(fs->lookup_cache, 0xff, sizeof(fs->lookup_cache));
+  addr = fs->ate_wra;
+
+  while (true)
+    {
+      /* Make a copy of 'addr' as it will be advanced by nvs_pref_ate() */
+
+      ate_addr = addr;
+      rc = nvs_prev_ate(fs, &addr, &ate);
+
+      if (rc)
+        {
+          return rc;
+        }
+
+      cache_entry = &fs->lookup_cache[nvs_lookup_cache_pos(ate.id)];
+
+      if (ate.id != 0xffffffff && *cache_entry == NVS_LOOKUP_CACHE_NO_ADDR &&
+          nvs_ate_valid(fs, &ate))
+        {
+          *cache_entry = ate_addr;
+        }
+
+      if (addr == fs->ate_wra)
+        {
+          break;
+        }
+    }
+
+  return 0;
+}
+
+static void nvs_lookup_cache_invalidate(struct nvs_fs *fs, uint32_t sector)
```

Comment:
```suggestion
static void nvs_lookup_cache_invalidate(FAR struct nvs_fs *fs, uint32_t sector)
```
here and other places